### PR TITLE
Implemented delete account

### LINKED ISSUE #213: Add ability to delete Transporter account in AccountEditingScreen
The functionality for this is already available in the database and database service.

- [x] Add button in AccountEditingScreen to delete the account if the Transporter wants to
- [x] This will NOT delete the Transporter's ATRs
- [x] Launch a dialog to confirm that the user really does want to delete their account

### PR DESCRIPTION
**Pull request by MummenRider**
Screen Shot 2021-04-02 at 12 23 48 PM

### CODE REVIEW

File: app/lib/core/view_models/account_edit_view_model.dart [hunk 1]
```diff
@@ -89,4 +93,29 @@ class AccountEditViewModel extends BaseViewModel {
   }
 
   void navigateToAccount() => _navigationService.pop();
+
+  Future<void> removeTransporter(String userId) async =>
```
Comment (holtzmak) [inline app/lib/core/view_models/account_edit_view_model.dart]: - [x] Is this used even?

File: app/lib/core/view_models/account_edit_view_model.dart [hunk 2]
```diff
@@ -89,4 +93,29 @@ class AccountEditViewModel extends BaseViewModel {
...
+      description: 'Do you really want to deactivate your account??',
+      confirmationText: 'Yes',
+      cancelText: 'No',
+    );
+
+    if (!dialogResponse.confirmed) return;
+    setState(ViewState.Busy);
+    try {
+      _navigationService.navigateTo(WelcomeScreen.route);
+      await _authenticationService.deleteAccount();
```
Comment (holtzmak) [inline app/lib/core/view_models/account_edit_view_model.dart:109]: Does the navigation back to welcome screen still occur when deleting the account fails? (You can test this by disconnecting from the internet before hitting the deactivate button)

**Review by holtzmak — CHANGES_REQUESTED** [review]
Could you take a screenshot of the button lol?

File: app/lib/core/view_models/account_edit_view_model.dart (hunk 2 above)
Comment (MummenRider) [reply app/lib/core/view_models/account_edit_view_model.dart:109]: Yes, it goes to welcome screen.

Comment (holtzmak) [reply app/lib/core/view_models/account_edit_view_model.dart:109]: Hm, I would prefer it not to back out if failure, but not really important to fix now.

**Review by holtzmak — APPROVED** [review]

**Pull request merged**